### PR TITLE
[wip] [videoplayer] Add video stream selection to core

### PR DESCRIPTION
**Pull request by ace20022**
This pr replaces a part of pr #8514
Remaining tasks
- [x] adapt default video stream opening
- [x] save video stream selection
- [x] Add angle selection for dvds

### CODE REVIEW

File: addons/resource.language.en_gb/resources/strings.po [hunk 1]
```diff
@@ -2090,7 +2090,11 @@ msgctxt "#470"
 msgid "Credits"
 msgstr ""
 
-#empty strings from id 471 to 473
+msgctxt "#471"
+msgid "Video stream"
+msgstr ""
+
+#empty strings from id 472 to 473
```
Comment (MartijnKaijser) [inline addons/resource.language.en_gb/resources/strings.po]: could you use a string ID above 10000 please? Thank you :)

File: xbmc/cores/VideoPlayer/DVDMessage.h [hunk 2]
```diff
@@ -185,6 +186,15 @@ class CDVDMsgPlayerSetAudioStream : public CDVDMsg
   int m_streamId;
 };
 
+class CDVDMsgPlayerSetVideoStream : public CDVDMsg
+{
+public:
+  CDVDMsgPlayerSetVideoStream(int streamId) : CDVDMsg(PLAYER_SET_VIDEOSTREAM) { m_streamId = streamId; }
+  int GetStreamId() { return m_streamId; }
```
Comment (Jalle19) [inline xbmc/cores/VideoPlayer/DVDMessage.h]: const

File: xbmc/cores/IPlayer.h [hunk 3]
```diff
@@ -290,6 +290,11 @@ class IPlayer
   virtual void SetAudioStream(int iStream){};
   virtual void GetAudioStreamInfo(int index, SPlayerAudioStreamInfo &info){};
 
+  virtual int GetVideoStream() { return -1; }
+  virtual int GetVideoStreamCount() { return 0; }
```
Comment (Jalle19) [inline xbmc/cores/IPlayer.h]: const here as well I'd assume

File: xbmc/video/dialogs/GUIDialogVideoSettings.cpp [hunk 4]
```diff
@@ -364,3 +384,53 @@ void CGUIDialogVideoSettings::InitializeSettings()
...
+    std::string strItem;
+    std::string strLanguage;
+
+    SPlayerVideoStreamInfo info;
+    g_application.m_pPlayer->GetVideoStreamInfo(i, info);
+
+    if (!g_LangCodeExpander.Lookup(info.language, strLanguage))
+      strLanguage = g_localizeStrings.Get(13205); // Unknown
+
+    if (info.name.length() == 0)
```
Comment (Jalle19) [inline xbmc/video/dialogs/GUIDialogVideoSettings.cpp]: empty()

File: xbmc/video/dialogs/GUIDialogVideoSettings.cpp [hunk 5]
```diff
@@ -364,3 +384,53 @@ void CGUIDialogVideoSettings::InitializeSettings()
...
+      strItem = strLanguage;
+    else
+      strItem = StringUtils::Format("%s - %s", strLanguage.c_str(), info.name.c_str());
+
+    if (info.videoCodecName.length() == 0)
```
Comment (Jalle19) [inline xbmc/video/dialogs/GUIDialogVideoSettings.cpp]: Same here

**Pull request closed without merging**